tests/validation/tests/v3_api/test_node.py

```diff
@@ -1,54 +1,961 @@
+    assert len(nodes) > 0
+    node_id = nodes[0].id
+    node = client.by_id_node(node_id)
+    node_name = node.nodeName
+
+    # add label on node
+    command = "label nodes " + node_name + " " + test_label + "=" + label_value
+    print(command)
+    execute_kubectl_cmd(command, False)
+    time.sleep(2)
+
+    # Label should be added
+    node = client.reload(node)
+    wait_for_condition(client, node, check_label_added(test_label), None, 10)
+    validate_label_set_on_node(client, node, test_label, label_value)
+
+    # edit label through kubectl
+    new_value = random_name()
+    command = "label nodes " + node_name + " " + \
+              test_label + "=" + new_value + " --overwrite"
+    print(command)
+    execute_kubectl_cmd(command, False)
+    node = client.reload(node)
+    time.sleep(2)
+
+    # New Label should be added
+    wait_for_condition(client, node, check_label_added(test_label), None, 10)
+    validate_label_set_on_node(client, node, test_label, label_value=new_value)
+
+    # remove label
+    node = client.reload(node)
+    node_labels = node.labels.data_dict()
+    del node_labels[test_label]
+    client.update(node, labels=node_labels)
+
+
+def test_node_label_kubectl_delete():
+    create_kubeconfig(cluster_detail["cluster"])
+    client = cluster_detail["client"]
+    cluster = cluster_detail["cluster"]
+    test_label = random_name()
+    label_value = random_name()
+    nodes = client.list_node(clusterId=cluster.id).data
+    assert len(nodes) > 0
+    node_id = nodes[0].id
+    node = client.by_id_node(node_id)
+    node_name = node.nodeName
+
+    # add label on node
+    command = "label nodes " + node_name + " " + test_label + "=" + label_value
+    print(command)
+    execute_kubectl_cmd(command, False)
+    time.sleep(2)
+
+    # Label should be added
+    wait_for_condition(client, node, check_label_added(test_label), None, 10)
+    validate_label_set_on_node(client, node, test_label, label_value)
+
+    # remove label through kubectl
+    command = " label node " + node_name + " " + test_label + "-"
+    execute_kubectl_cmd(command, False)
+    time.sleep(2)
+
+    # label should be deleted
+    wait_for_condition(client, node, check_label_removed(test_label), None, 10)
+    validate_label_deleted_on_node(client, node, test_label)
+
+
+def test_node_label_combination_1():
+    create_kubeconfig(cluster_detail["cluster"])
+    """Add via kubectl, Delete via API, Add via kubectl"""
+    client = cluster_detail["client"]
+    cluster = cluster_detail["cluster"]
+    test_label = random_name()
+    label_value = random_name()
+    nodes = client.list_node(clusterId=cluster.id).data
+    assert len(nodes) > 0
+    node_id = nodes[0].id
+    node = client.by_id_node(node_id)
+    node_name = node.nodeName
+
+    # delete if label is present
+    node_labels = node.labels.data_dict()
+    if test_label in node_labels:
+        del node_labels[test_label]
+        client.update(node, labels=node_labels)
+
```

Comment:
This may not be needed since we use random label names now.

Reply:
done